From 7f56dd9b627b32ed762ed878da738d4a1128113e Mon Sep 17 00:00:00 2001
From: adabaca <ada@olivestudio.ro>
Date: Fri, 26 Apr 2024 08:49:59 +0300
Subject: [PATCH] Fixed translation not working.

---
 lib/class-performant-translations.php | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/class-performant-translations.php b/lib/class-performant-translations.php
index 2e1255480..e6aa0a83a 100644
--- a/lib/class-performant-translations.php
+++ b/lib/class-performant-translations.php
@@ -19,14 +19,14 @@ class Performant_Translations {
 	public static function init() {
 		if ( ! class_exists( 'WP_Translation_Controller' ) ) {
 			// WordPress < 6.5
-			add_filter( 'override_load_textdomain', array( __CLASS__, 'load_textdomain' ), 100, 4 );
+			add_filter( 'override_load_textdomain', array( __CLASS__, 'load_textdomain' ), 9, 4 );
 			add_filter( 'override_unload_textdomain', array( __CLASS__, 'unload_textdomain' ), 100, 3 );
 
 			add_action( 'init', array( __CLASS__, 'set_locale' ) );
 			add_action( 'change_locale', array( __CLASS__, 'change_locale' ) );
 		} else {
 			// WordPress 6.5+
-			add_filter( 'load_translation_file', array( __CLASS__, 'load_translation_file' ), 100, 2 );
+			add_filter( 'load_translation_file', array( __CLASS__, 'load_translation_file' ), 9, 2 );
 		}
 
 		// Any version.
-- 
2.34.1

