From 594df1ea97d914c982fad8373a32e0923d55231e Mon Sep 17 00:00:00 2001
From: Fabian Marz <fabian@netzstrategen.com>
Date: Tue, 20 Jun 2023 12:02:08 +0200
Subject: [PATCH] Fixed real-media-library CPU performance when loading
 folders.

---
 inc/attachment/Structure.php | 15 +++++----------
 1 file changed, 5 insertions(+), 10 deletions(-)

diff --git a/inc/attachment/Structure.php b/inc/attachment/Structure.php
index f73252e20..c48ca5472 100644
--- a/inc/attachment/Structure.php
+++ b/inc/attachment/Structure.php
@@ -208,7 +208,7 @@ class Structure implements \MatthiasWeb\RealMediaLibrary\api\IStructure {
                 if (isset($creatable)) {
                     $result = \call_user_func_array([$creatable, 'instance'], [$value]);
                     if (is_rml_folder($result)) {
-                        $this->parsed[] = $result;
+                        $this->parsed[\intval($value->id)] = $result;
                     }
                 }
             }
@@ -228,6 +228,7 @@ class Structure implements \MatthiasWeb\RealMediaLibrary\api\IStructure {
         $this->parsed = apply_filters('RML/Tree/Parsed', $this->parsed);
         // Create the tree
         $folder = null;
+        $cats_tree = [];
         foreach ($this->parsed as $key => $category) {
             $parent = $category->getParent();
             if ($parent > -1) {
@@ -235,11 +236,7 @@ class Structure implements \MatthiasWeb\RealMediaLibrary\api\IStructure {
                 if ($folder !== null) {
                     $folder->addChildren($category);
                 }
-            }
-        }
-        $cats_tree = [];
-        foreach ($this->parsed as $category) {
-            if ($category->getParent() <= -1) {
+            } elseif ($parent <= -1) {
                 $cats_tree[] = $category;
             }
         }
@@ -251,10 +248,8 @@ class Structure implements \MatthiasWeb\RealMediaLibrary\api\IStructure {
         if (!$nullForRoot && $id === -1) {
             return \MatthiasWeb\RealMediaLibrary\folder\Root::getInstance();
         }
-        foreach ($this->getParsed() as $folder) {
-            if ($folder->getId() === $id) {
-                return $folder;
-            }
+        if ($found = $this->getParsed()[$id] ?? NULL) {
+          return $found;
         }
         /**
          * When a folder is not found by an absolute path this filter is
-- 
2.30.2

