From 16be4aa08895f0fbe1d3469add0b097d8e0d97d9 Mon Sep 17 00:00:00 2001
From: Fabian Marz <fabian@netzstrategen.com>
Date: Wed, 21 Jun 2023 15:13:16 +0200
Subject: [PATCH] Fixed real-media-library CPU performance when loading
 folders.

---
 inc/attachment/Structure.php | 15 +++++----------
 1 file changed, 5 insertions(+), 10 deletions(-)

diff --git a/inc/attachment/Structure.php b/inc/attachment/Structure.php
index 0c6102410..acec4520d 100644
--- a/inc/attachment/Structure.php
+++ b/inc/attachment/Structure.php
@@ -186,7 +186,7 @@ class Structure implements IStructure
                 if (isset($creatable)) {
                     $result = \call_user_func_array([$creatable, 'instance'], [$value]);
                     if (\is_rml_folder($result)) {
-                        $this->parsed[] = $result;
+                        $this->parsed[\intval($value->id)] = $result;
                     }
                 }
             }
@@ -206,6 +206,7 @@ class Structure implements IStructure
         $this->parsed = \apply_filters('RML/Tree/Parsed', $this->parsed);
         // Create the tree
         $folder = null;
+        $cats_tree = [];
         foreach ($this->parsed as $key => $category) {
             $parent = $category->getParent();
             if ($parent > -1) {
@@ -213,11 +214,7 @@ class Structure implements IStructure
                 if ($folder !== null) {
                     $folder->addChildren($category);
                 }
-            }
-        }
-        $cats_tree = [];
-        foreach ($this->parsed as $category) {
-            if ($category->getParent() <= -1) {
+            } elseif ($parent <= -1) {
                 $cats_tree[] = $category;
             }
         }
@@ -230,10 +227,8 @@ class Structure implements IStructure
         if (!$nullForRoot && $id === -1) {
             return Root::getInstance();
         }
-        foreach ($this->getParsed() as $folder) {
-            if ($folder->getId() === $id) {
-                return $folder;
-            }
+        if ($found = $this->getParsed()[$id] ?? NULL) {
+            return $found;
         }
         /**
          * When a folder is not found by an absolute path this filter is
-- 
2.25.1

