From d9866e16555f0d5f89fe4bb268b7ad8a84ea9dff Mon Sep 17 00:00:00 2001
From: Nabi <artsanaat@gmail.com>
Date: Thu, 29 Feb 2024 14:35:09 +0330
Subject: [PATCH] Fixed search-exclude plugin overwriting the excluded ids.

---
 lib/class-plugin.php | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/class-plugin.php b/lib/class-plugin.php
index cb586419c..9126b9310 100644
--- a/lib/class-plugin.php
+++ b/lib/class-plugin.php
@@ -301,7 +301,8 @@ final class Plugin {
 		$exclude = apply_filters( 'searchexclude_filter_search', $exclude, $query );
 
 		if ( $exclude ) {
-			$query->set( 'post__not_in', array_merge( array(), $this->get_excluded() ) );
+      $excluded_ids = $query->query_vars['post__not_in'] ?? [];
+      $query->set('post__not_in', array_unique(array_merge($excluded_ids, $this->get_excluded())));
 		}
 
 		return $query;
-- 
2.15.0

