From 2ae2aa5ea6842feb5ad525221072035bb3f26d2f Mon Sep 17 00:00:00 2001
From: joram77 <56529054+joram77@users.noreply.github.com>
Date: Wed, 9 Jun 2021 18:36:43 +0200
Subject: [PATCH] Fixed PHP notice when empty groups appear on checkout.

---
 woochimp.class.php | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/woochimp.class.php b/woochimp.class.php
index 0fe680b..85e25e6 100644
--- a/woochimp.class.php
+++ b/woochimp.class.php
@@ -4790,7 +4790,9 @@ public function add_permission_question()
         // Display the html
         if ($already_subscribed === false || ($already_subscribed && ($this->opt['woochimp_hide_checkbox'] == '2' || ($this->opt['woochimp_hide_checkbox'] == '3' && !empty($groups['data']) && !empty($groups['html']))))) {
             echo $checkbox_block;
-            echo $groups['html'];
+            if (!empty($groups['html'])) {
+                echo $groups['html'];
+            }
         }
 
         // Load assets

