From 1ec2c118b0cefefd19b322d1fd87d1cfdebfae67 Mon Sep 17 00:00:00 2001
From: Daniel Kudwien <daniel@netzstrategen.com>
Date: Thu, 30 Jan 2025 16:41:50 +0100
Subject: [PATCH] Fixed wc_catalog_restrictions translations are loaded too
 early.

---
 woocommerce-catalog-visibility-options.php | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/woocommerce-catalog-visibility-options.php b/woocommerce-catalog-visibility-options.php
index 71b39edc9..682d96614 100644
--- a/woocommerce-catalog-visibility-options.php
+++ b/woocommerce-catalog-visibility-options.php
@@ -44,7 +44,9 @@ function wc_catalog_visibility_options_is_woocommerce_active(): bool {
 if ( wc_catalog_visibility_options_is_woocommerce_active() ) {
 	require 'wc-catalog-visibility-compatibility.php';
 
+	add_action( 'init', function () {
 	load_plugin_textdomain( 'wc_catalog_restrictions', null, dirname( plugin_basename( __FILE__ ) ) . '/languages' );
+	} );
 
 	// Declare support for features
 	add_action( 'before_woocommerce_init', function () {
-- 
2.25.1

