From 1ca921c4c38b46bfb046d972826e67560da3a2d3 Mon Sep 17 00:00:00 2001
From: Bogdan Arizancu <bogdan@netzstrategen.com>
Date: Mon, 24 Jan 2022 22:21:20 +0200
Subject: [PATCH] Fixed PHP notice caused by woochimp when empty groups appear
 on checkout.

---
 woochimp.class.php | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/woochimp.class.php b/woochimp.class.php
index 7fde32390..a3acff694 100644
--- a/woochimp.class.php
+++ b/woochimp.class.php
@@ -4790,7 +4790,9 @@ class WooChimp
         // Display the html
         if ($already_subscribed === false || ($already_subscribed && ($this->opt['woochimp_hide_checkbox'] == '2' || ($this->opt['woochimp_hide_checkbox'] == '3' && !empty($groups['data']) && !empty($groups['html']))))) {
             echo $checkbox_block;
-            echo $groups['html'];
+            if (!empty($groups['html'])) {
+                echo $groups['html'];
+            }
         }
 
         // Load assets
-- 
2.30.1 (Apple Git-130)

