From e5585e344cfd434398753a4850539ab1e08a80ba Mon Sep 17 00:00:00 2001
From: creative-andrew <andres@netzstrategen.com>
Date: Fri, 10 May 2024 13:13:50 +0200
Subject: [PATCH] fix: deprecate function call in wp-graphql-woocommerce

---
 includes/connection/class-customers.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/connection/class-customers.php b/includes/connection/class-customers.php
index 56276905cf..c2c010f417 100644
--- a/includes/connection/class-customers.php
+++ b/includes/connection/class-customers.php
@@ -201,7 +201,7 @@ class Customers {
 	 * @return array
 	 */
 	public static function upgrade_models( $connection, $resolver ) {
-		if ( 'customers' === $resolver->getInfo()->fieldName ) {
+		if ( 'customers' === $resolver->get_info()->fieldName ) {
 			$nodes = [];
 			$edges = [];
 			foreach ( $connection['nodes'] as $node ) {
-- 
2.24.4

